### PR TITLE
Source added to split, sorting of graph, change in crossfold, bug fixes

### PR DESCRIPTION
**Pull request by nomisto**
- Added source of edges to train, test, validation set
- Fixed a bug where ID's were splitted by underscore (old ids), not by ':' (CURIES)
- fixed a gui bug "--tn_edges" does not exist
- Before the graph is written to the files, it is sorted for reproducibility
- Improved memory efficiency of cross fold: each fold is now serialized after creation. Previous: All folds created -> all folds written
- Reworked trainTestSetWriter.py

Finalizes #4

### CODE REVIEW

File: src/openbiolink/edge.py [hunk 1]
```diff
@@ -1,27 +1,41 @@
 from openbiolink.edgeType import EdgeType
+from openbiolink.namespace import Namespace
 
 
 class Edge:
-    def __init__(self, id1: str, type: EdgeType, id2: str, source: "", qScore=None, sourcedb=None):
```
Comment (cthoyt) [inline src/openbiolink/edge.py]: why not just include instances of `openbiolink.node.Node`?

**Comment by nomisto** [comment]
@cthoyt  I've tried rewriting the train/test split module to not use global parameters and instead function/instance parameters. But this gets out of hand very quickly, the COL_NAMES_* and *_COL_NAME are basically the values of the pandas headers and some function's require multiple of then.
Passing them all by function arguments would give us very long function definitions.
Mabe you know a way to make this easier to read/implement.

https://github.com/OpenBioLink/OpenBioLink/blob/24632aa0e26e318aa994bc823a0f7b65d225ef52/src/openbiolink/openBioLink.py#L193-L195

https://github.com/OpenBioLink/OpenBioLink/blob/24632aa0e26e318aa994bc823a0f7b65d225ef52/src/openbiolink/train_test_set_creation/trainTestSplitCreation.py#L89-L95

File: src/openbiolink/edge.py (hunk 1 above)
Comment (nomisto) [reply src/openbiolink/edge.py]: Did not see that :D thanks!

File: src/openbiolink/edge.py [hunk 2]
```diff
@@ -1,41 +1,40 @@
...
+            return iter([self.resolve(self.node1), self.type,
+                         self.resolve(self.node2), "", self.sourcedb])
 
     def to_sub_rel_obj_list(self):
-        return iter([self.id1, self.type, self.id2, self.sourcedb])
+        return iter([self.node1.id, self.type, self.node2.id, self.sourcedb])
+
+    @staticmethod
+    def resolve(node):
+        return node.namespace.resolve(node.id)
```
Comment (cthoyt) [inline src/openbiolink/edge.py]: would be nice just to make a property of the node

```python
class Node:

    ...

    @property
    def resolved_id(self):
        return self.namespace.resolve(self.id)
```

**Pull request merged**